src/v/ssx/thread_worker.h

```diff
@@ -95,13 +140,11 @@ class worker_task final : public task_base {
 };
 
 class thread_worker {
-    static constexpr eventfd_t RUNNABLE_INIT = 0;
-    static constexpr eventfd_t RUNNABLE_READY = 1;
-    static constexpr eventfd_t RUNNABLE_STOP = 2;
-
 public:
-    thread_worker()
-      : _alien{ss::engine().alien()}
+    explicit thread_worker(thread_worker_config* config)
```

Comment:
Could the config lifetime be simplified by being wired through `start` rather than held by a pointer at construction?

Reply:
Good idea, much cleaner 😄